nsqd/stats.go

```diff
@@ -145,6 +145,37 @@ func (n *NSQD) GetStats() []TopicStats {
 	return topics
 }
 
+func (n *NSQD) GetTopicStats(topic string) TopicStats {
+	n.RLock()
+	var realTopic *Topic
+	for _, t := range n.topicMap {
+		if topic == t.name {
+			realTopic = t
+			break
+		}
+	}
+	n.RUnlock()
+	realTopic.RLock()
+	realChannels := make([]*Channel, 0, len(realTopic.channelMap))
+	for _, c := range realTopic.channelMap {
+		realChannels = append(realChannels, c)
+	}
+	realTopic.RUnlock()
+	sort.Sort(ChannelsByName{realChannels})
+	channels := make([]ChannelStats, 0, len(realChannels))
+	for _, c := range realChannels {
+		c.RLock()
+		clients := make([]ClientStats, 0, len(c.clients))
+		for _, client := range c.clients {
```

Comment:
Hmm ... we'd need to add a new param, to omit the client connection details from the stats response, to keep compatibility with anything out there that depends on the full stats responses. Luckily, the `/stats` endpoint will ignore params it is not expecting, so new nsqadmin will still work with old nsqd, it'll just get more info than it needs.

(The topic and channel params to `/stats` have existed for quite a while, since nsqd 0.3.6)